internal/gatewayapi/route.go

```diff
@@ -1160,3 +1165,96 @@ func getIREndpointsFromEndpointSlice(endpointSlices []*discoveryv1.EndpointSlice
 
 	return endpoints
 }
+
+func GetTargetBackendReference(backendRef gwapiv1a1.BackendRef, namespace string) gwapiv1a1.PolicyTargetReferenceWithSectionName {
+	ref := gwapiv1a1.PolicyTargetReferenceWithSectionName{
+		PolicyTargetReference: gwapiv1a1.PolicyTargetReference{
+			Group:     *backendRef.Group,
+			Kind:      *backendRef.Kind,
+			Name:      backendRef.Name,
+			Namespace: NamespacePtr(NamespaceDerefOr(backendRef.Namespace, namespace)),
+		},
+		SectionName: SectionNamePtr(strconv.Itoa(int(*backendRef.Port))),
+	}
+	return ref
+}
+
+func TargetMatched(policy gwapiv1a1.BackendTLSPolicy, target gwapiv1a1.PolicyTargetReferenceWithSectionName) bool {
+
+	policyTarget := policy.Spec.TargetRef
+
+	if target.Group == policyTarget.Group &&
+		target.Kind == policyTarget.Kind &&
+		target.Name == policyTarget.Name &&
+		NamespaceDerefOr(policyTarget.Namespace, policy.Namespace) == string(*target.Namespace) {
+		if policyTarget.SectionName != nil && *policyTarget.SectionName != *target.SectionName {
+			return false
+		}
+		return true
+	}
+	return false
+}
+
+func getBackendTLSPolicy(policies []*gwapiv1a1.BackendTLSPolicy, target gwapiv1a1.PolicyTargetReferenceWithSectionName) *gwapiv1a1.BackendTLSPolicy {
+	for _, policy := range policies {
+		if TargetMatched(*policy, target) {
+			return policy
+		}
+	}
+	return nil
+}
+
+func getBackendTLSBundle(policies []*gwapiv1a1.BackendTLSPolicy, configmaps []*corev1.ConfigMap, secrets []*corev1.Secret, targetBackend gwapiv1a1.PolicyTargetReferenceWithSectionName) *ir.TLSBundle {
+
+	backendTLSPolicy := getBackendTLSPolicy(policies, targetBackend)
+
+	if backendTLSPolicy == nil {
+		return nil
+	}
+
+	tlsBundle := &ir.TLSBundle{}
+
+	caRefMap := make(map[string]string)
+
+	for _, caRef := range backendTLSPolicy.Spec.TLS.CACertRefs {
+		caRefMap[string(caRef.Name)] = string(caRef.Kind)
+	}
+
+	ca := ""
+
+	for _, cmap := range configmaps {
+		if kind, ok := caRefMap[cmap.Name]; ok && kind == cmap.Kind {
+			if crt, dataOk := cmap.Data["ca.crt"]; dataOk {
+				if ca != "" {
+					ca += "\n"
+				}
+				ca += crt
+			}
+		}
+	}
+
+	for _, secret := range secrets {
+		if kind, ok := caRefMap[secret.Name]; ok && kind == secret.Kind {
+			if crt, dataOk := secret.Data["ca.crt"]; dataOk {
```

Comment:
Will ca be stored in a secret？

Reply:
I'm not sure about this. CA is not a secret and should be stored in a configMap, as this [example](https://gateway-api.sigs.k8s.io/guides/tls/#using-explicit-ca-certificates) shows.

```yaml
apiVersion: gateway.networking.k8s.io/v1alpha2
kind: BackendTLSPolicy
metadata:
  name: tls-upstream-auth
spec:
  targetRef:
    kind: Service
    name: auth-service
    group: ""
  tls:
    caCertRefs:
      - kind: ConfigMapReference
        name: auth-cert
        group: ""
    hostname: auth.example.com
```

Reply:
ok, will ignore ca files in the secrets